client/nginx.go

```diff
@@ -386,6 +388,10 @@ func (client *NginxClient) AddHTTPServer(upstream string, server UpstreamServer)
 		return fmt.Errorf("failed to add %v server to %v upstream: server already exists", server.Server, upstream)
 	}
 
+	if server.MaxFails == nil {
```

Comment:
hm. why? if you add `omitempty` and if maxFails is nil, that field will not be present in JSON.